### PR TITLE
feat: upgrade Open Zaak APIs

### PR DESCRIPTION
**Pull request by edgarvonk**
Upgraded Open Zaak APIs in ZAC including the corresponding Java client code from the updated OpenAPI specs. #major

Solves PZ-1019

### CODE REVIEW

File: scripts/docker-compose/imports/openzaak-database/database/1-setup-zac-config.sql [hunk 1]
```diff
@@ -1,7 +1,7 @@
 CREATE EXTENSION IF NOT EXISTS "uuid-ossp";
 
-INSERT INTO catalogi_catalogus (_admin_name, uuid, domein, rsin, contactpersoon_beheer_naam, contactpersoon_beheer_telefoonnummer, contactpersoon_beheer_emailadres, _etag) VALUES
-    ('zac', '8225508a-6840-413e-acc9-6422af120db1', 'ALG', '002564440', 'ZAC Test Catalogus', '06-12345678', 'noreply@example.com', '_etag');
+INSERT INTO catalogi_catalogus (naam, uuid, domein, rsin, contactpersoon_beheer_naam, contactpersoon_beheer_telefoonnummer, contactpersoon_beheer_emailadres, _etag, begindatum_versie, versie) VALUES
+    ('zac', '8225508a-6840-413e-acc9-6422af120db1', 'ALG', '002564440', 'ZAC Test Catalogus', '06-12345678', 'noreply@example.com', '_etag', NULL, '');
```
Comment (bas-info-nl) [inline scripts/docker-compose/imports/openzaak-database/database/1-setup-zac-config.sql:4]: We don't want to add a version nor start date? Will this not much things up for future versions? I suppose we'll find out...

File: src/itest/kotlin/nl/lifely/zac/itest/ZakenRESTServiceTest.kt [hunk 2]
```diff
@@ -67,4 +67,11 @@ class ZakenRESTServiceTest : BehaviorSpec({
             }
         }
     }
+    // TODO: test for Intake afronden
```
Comment (bas-info-nl) [inline src/itest/kotlin/nl/lifely/zac/itest/ZakenRESTServiceTest.kt:70]: adding todo's is not the best practice, but maybe ok here?

File: src/main/resources/openapi-generator-templates/pojo.mustache
Comment (bas-info-nl) [inline src/main/resources/openapi-generator-templates/pojo.mustache:1]: I'm unable to review this code properly. Or would you like me to dive into it?

**Review by bas-info-nl — APPROVED** [review]

File: scripts/docker-compose/imports/openzaak-database/database/1-setup-zac-config.sql (hunk 1 above)
Comment (edgarvonk) [reply scripts/docker-compose/imports/openzaak-database/database/1-setup-zac-config.sql:4]: Not for now. Begin datum is optional and I have not idea what is normally put in the new versie param. Also: this is straight from the OpenZaak upgrade scripts. They put an empty string in here. So for now this is ok I think.

File: src/itest/kotlin/nl/lifely/zac/itest/ZakenRESTServiceTest.kt (hunk 2 above)
Comment (edgarvonk) [reply src/itest/kotlin/nl/lifely/zac/itest/ZakenRESTServiceTest.kt]: this is a draft PR. it is not ready for review yet!

File: src/main/resources/openapi-generator-templates/pojo.mustache
Comment (edgarvonk) [reply src/main/resources/openapi-generator-templates/pojo.mustache:1]: I have indicated with a code comment ("Lifely/INFO workaround") what I changed from the original Mustache template of OpenAPI Generator. That's the only part you need to review. :-)

**Review by Jorann — APPROVED** [review]
Looks good and still seems to work

**Pull request merged**